cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -655,62 +469,37 @@ private boolean getClusterLoad(HttpServletRequest request, HttpServletResponse r
+    String resourceString = KafkaCruiseControlServletUtils.resourceString(request);
     try {
-      String resourceString = request.getParameter(RESOURCE_PARAM);
-      try {
-        if (resourceString == null) {
-          resourceString = DEFAULT_PARTITION_LOAD_RESOURCE;
-        }
-        resource = Resource.valueOf(resourceString.toUpperCase());
-      } catch (IllegalArgumentException iae) {
-        String errorMsg = String.format("Invalid resource type %s. The resource type must be one of the following: "
-                                        + "CPU, DISK, NW_IN, NW_OUT", resourceString);
-        StringWriter sw = new StringWriter();
-        iae.printStackTrace(new PrintWriter(sw));
-        setErrorResponse(response, sw.toString(), errorMsg, SC_BAD_REQUEST, json);
-        // Close session
-        return true;
-      }
-      String startMsString = request.getParameter(START_MS_PARAM);
-      String endMsString = request.getParameter(END_MS_PARAM);
-      startMs = startMsString == null ? -1L : Long.parseLong(startMsString);
-      endMs = endMsString == null ? System.currentTimeMillis() : Long.parseLong(endMsString);
-      String entriesString = request.getParameter(ENTRIES_PARAM);
-      entries = entriesString == null ? Integer.MAX_VALUE : Integer.parseInt(entriesString);
-      String topicString = request.getParameter(TOPIC_PARAM);
-      if (topicString != null) {
-        topic = Pattern.compile(topicString);
-      } else {
-        topic = null;
-      }
-      String partitionString = request.getParameter(PARTITION_PARAM);
-      if (partitionString == null) {
-        partitionLowerBoundary = Integer.MIN_VALUE;
-        partitionUpperBoundary = Integer.MAX_VALUE;
-      } else if (!partitionString.contains("-")) {
-        partitionLowerBoundary = Integer.parseInt(partitionString);
-        partitionUpperBoundary = partitionLowerBoundary;
-      } else {
-        String [] partitionBoundaries = partitionString.split("-");
-        partitionLowerBoundary = Integer.parseInt(partitionBoundaries[0]);
-        partitionUpperBoundary = Integer.parseInt(partitionBoundaries[1]);
-      }
+      resource = Resource.valueOf(resourceString.toUpperCase());
+    } catch (IllegalArgumentException iae) {
+      String errorMsg = String.format("Invalid resource type %s. The resource type must be one of the following: "
+                                      + "CPU, DISK, NW_IN, NW_OUT", resourceString);
+      handleParameterParseException(iae, response, errorMsg, json);
+      // Close session
+      return true;
+    }
+
+    try {
+      Long startMsValue = KafkaCruiseControlServletUtils.startMs(request);
+      startMs = startMsValue == null ? -1L : startMsValue;
+      Long endMsValue = KafkaCruiseControlServletUtils.endMs(request);
+      endMs = endMsValue == null ? System.currentTimeMillis() : endMsValue;
+      partitionLowerBoundary = KafkaCruiseControlServletUtils.partitionLowerBoundary(request);
```

Comment:
I think it'll be good to convert this into a single method where an upper and lower bound is returned.